csrc/tensor_view.cpp

```diff
@@ -1198,17 +1198,43 @@ void TensorView::clearReductionIterDomains() {
       getLeafDomain() == getRootDomain(),
       "should not call clearReductionIterDomains on already transformed TensorDomains");
 
+  const std::vector<IterDomain*>& root = getRootDomain();
+  const std::vector<IterDomain*>& alloc = getMaybeAllocationDomain();
+
+  NVF_ERROR(
+      std::is_permutation(root.begin(), root.end(), alloc.begin(), alloc.end()),
+      "should not call clearReductionIterDomains on transformed allocation domain");
```

Comment:
Won't this break too soon with @zasdfgbnm 's TMA work?

Reply:
This may just be my ignorance on when clearReductionIterDomain is called.

Reply:
I think the assumption in this API call is that

`"should not call clearReductionIterDomains on already transformed TensorDomains"`

It does have an existing check on `leafdomain` being the same as with `rootdomain`.. which I think hints that there's no TMADomain neither.

Reply:
Got it. It's currently used only on outputs of fusion segments, for which the assumption holds.